### PR TITLE
Cleanup pom files

### PR DESCRIPTION
**Pull request by davidegrohmann**
- remove java language version from ndp poms
- use ndp-parent pom in sub-modules
- remove explicit dependency versions from poms

### CODE REVIEW

File: community/ndp/pom.xml [hunk 1]
```diff
@@ -14,9 +14,11 @@
     <license-text.header>GPL-3-header.txt</license-text.header>
     <licensing.prepend.text>notice-gpl-prefix.txt</licensing.prepend.text>
     <componentversion.skip>true</componentversion.skip>
+    <ascii.doctor.version>1.5.2</ascii.doctor.version>
+    <jsoup.version>1.8.1</jsoup.version>
   </properties>
 
-  <groupId>org.neo4j.build</groupId>
+  <groupId>org.neo4j.ndp</groupId>
```
Comment (jakewins) [inline community/ndp/pom.xml:21]: The NDP parent is just a build artifact, similar to the 'community' pom, not something one can depend on. In other places where we've had these build-only poms we've kept them in the 'build' space. Is there a case for moving away from that here?

File: community/ndp/kernelextension/pom.xml [hunk 2]
```diff
@@ -5,10 +5,10 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.neo4j</groupId>
-    <artifactId>parent</artifactId>
+    <groupId>org.neo4j.ndp</groupId>
+    <artifactId>neo4j-data-protocol-parent</artifactId>
```
Comment (jakewins) [inline community/ndp/kernelextension/pom.xml:9]: This breaks from how we've done parents elsewhere - not sure if that's good or bad, but it's different :)

Other modules in the project use the `org.neo4j:parent` parent, that way there's a single parent for all the modules. See for instance: https://github.com/neo4j/neo4j/blob/2.3/community/function/pom.xml and note that that project is under the 'community' multi-module.

I think that the philosophy has been that the multi-module projects organize the modules in a tree, but they all inherit the same parent for ease of configuration.

Comment (davidegrohmann) [inline community/ndp/kernelextension/pom.xml:9]: If you look at https://github.com/neo4j/neo4j/blob/2.3/community/cypher/cypher/pom.xml it depends on the `cypher-parent`. So this is not very different wrt other modules we have in the code base.

Comment (jakewins) [inline community/ndp/kernelextension/pom.xml:9]: Hmm, I hadn't seen that. Fair enough. I don't feel very strongly about it, and it's more intuitive that the parent directory also has the direct parent. 

Also, this is all kind of related to the devteam mail about module structure. Happy to merge this as-is.

File: community/ndp/pom.xml (hunk 1 above)
Comment (davidegrohmann) [inline community/ndp/pom.xml:21]: It looked like to be more consistent this way, but we can revert it if it is a problem.

**Pull request merged**